### ISSUE TITLE
Should TypeAliasType check that it is parametrized correctly?

### ISSUE THREAD

**Issue by adriangb** [python/typing_extensions#239]
The new TypeAliasType does not seem to check its generic parameters at all:

```python
from typing import TypeVar
from typing_extensions import TypeAliasType

T = TypeVar('T')

Bad1 = TypeAliasType('Bad1', list[T])
Bad1[int]  # should error

Bad2 = TypeAliasType('Bad2', list[T], type_parameters=(T,))
Bad2[int, int]  # should error
```

This is not enforced for most built-in types it seems (`list[int, int]`) but it is for classes using `Generic[T]`.

I suppose this is the same behavior in CPython but I'm bringing it up here for discussion.

**Comment by JelleZijlstra** [comment]
Yes, this is intentional and matches the behavior of `types.GenericAlias`. In general I want to push enforcement of complicated restrictions out of the Python core and to third-party type checkers.
